### PR TITLE
Fix "xiangqihouse" to Align with Actual Xiangqi Bughouse

### PR DESCRIPTION
**Pull request by PraseodymiumSpike**
better follows the information in this message: https://discord.com/channels/634298688663191582/634327888275177482/1136516400345780264

### CODE REVIEW

**Comment by ianfab** [comment]
Thanks. The black and white soldier look interchanged, but other than that looks fine.

File: src/variants.ini [hunk 1]
```diff
@@ -339,6 +339,12 @@ capturesToHand = true
 dropChecks = false
 whiteDropRegion = *1 *2 *3 *4 *5
 blackDropRegion = *6 *7 *8 *9 *10
+mobilityRegionWhiteFers = d1 f1 e2 d3 f3
+mobilityRegionBlackFers = d8 f8 e9 d10 f10
+mobilityRegionWhiteElephant = c1 g1 a3 e3 i3 c5 g5
+mobilityRegionBlackElephant = c6 g6 a8 e8 i8 c10 g10
+mobilityRegionWhiteSoldier = a4 a5 c4 c5 e4 e5 g4 g5 *6 *7 *8 *9 *10
```
Comment (ianfab) [inline src/variants.ini:346]: Upon a closer look it seems like the i file squares are missing here.

Comment (PraseodymiumSpike) [reply src/variants.ini]: It should be fixed now. Thank you! Hopefully, this is the last error.

**Comment by PraseodymiumSpike** [comment]
I checked everything again, and I think it's correct now.

**Comment by gbtami** [comment]
Shouldn't this be applied to Suppy chess (Xiangqi bughouse) as well?

**Comment by PraseodymiumSpike** [comment]
@gbtami Yes, it should be applied to Supply Chess.

**Comment by ianfab** [comment]
Are you planning to update the PR with the changes for supply chess as well?

**Comment by PraseodymiumSpike** [comment]
> Are you planning to update the PR with the changes for supply chess as well?


@ianfab Sorry, I wasn't aware before that Supply Chess was defined in the code. I don't know how to program in C++, though, and I don't want to mess anything up or miss something, so I think that perhaps, you should fix Supply Chess yourself. I guess I'll open an issue for it.

**Pull request merged**